src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -135,6 +135,10 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                         apiEvent.TenantId = authenticationResult.TenantId;
                         apiEvent.AccountId = authenticationResult.UniqueId;
                         apiEvent.WasSuccessful = true;
+                        if (AuthenticationRequestParameters.ApiId == ApiEvent.ApiIds.AcquireTokenSilent)
```

Comment:
Please move this to SilentRequest subclass. Typically when there is a class hierarchy, having code that is like

```
if (subtype is X) then do stuff
```

is a code smell. It suggests that either the class hierarchy is wrong or that there needs to be more virtual methods (hooks) in the hierarchy. In this case it's pretty simple, I think you can move this to SilentRequest